docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,355 @@
+  verbs: ['*']
+- apiGroups: [batch]
+  resources: [jobs]
+  verbs: ['*']
+- apiGroups: [config.openshift.io]
+  resources: [clusterversions]
+  verbs: [get, list, watch]
+- apiGroups: [monitoring.coreos.com]
+  resources: [prometheuses]
+  verbs: ['*']
+- apiGroups: [monitoring.coreos.com]
+  resources: [servicemonitors]
+  verbs: ['*']
+- apiGroups: [networking.k8s.io]
+  resources: [ingresses]
+  verbs: ['*']
+- apiGroups: [oauth.openshift.io]
+  resources: [oauthclients]
+  verbs: [create, delete, get, list, patch, update, watch]
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: ['*']
+  verbs: ['*']
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: ['*']
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: ['*']
+- apiGroups: [route.openshift.io]
+  resources: [routes]
+  verbs: ['*']
+- apiGroups: [route.openshift.io]
+  resources: [routes/custom-host]
+  verbs: ['*']
+- apiGroups: [template.openshift.io]
+  resources: [templateconfigs]
+  verbs: ['*']
+- apiGroups: [template.openshift.io]
+  resources: [templateinstances]
+  verbs: ['*']
+- apiGroups: [template.openshift.io]
+  resources: [templates]
+  verbs: ['*']
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: argocd-installer-role
+  namespace: argocd
+rules:
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [create]
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [get, list, watch, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+- apiGroups: [""]
+  resources: [configmaps]
+  verbs: [create]
+- apiGroups: [""]
+  resources: [configmaps]
+  verbs: [get, list, watch, update, patch, delete]
+  resourceNames: [argocd-operator-manager-config]
+- apiGroups: [""]
+  resources: [services]
+  verbs: [create]
+- apiGroups: [""]
+  resources: [services]
+  verbs: [get, list, watch, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager-metrics-service]
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [create]
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [get, list, watch, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+# Creation of ClusterRoleBinding using the cluster-admin ClusterRole in non-production environments
+
+```yaml
+# Create ClusterRole
+kubectl apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: my-cluster-extension-installer-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: my-cluster-extension-service-account
+  namespace: my-cluster-extension-namespace
+EOF
+
+```
+kubectl create clusterrolebinding my-cluster-extension-installer-role-binding \
+  --clusterrole=cluster-admin \
+  --serviceaccount=my-cluster-extension-namespace:my-cluster-installer-service-account
+```
```

Comment:
This is not rendering correctly. Also you might want to use 
```sh
command
```

Reply:
fixed